editor.planx.uk/src/@planx/components/Calculate/Editor.tsx

```diff
@@ -36,12 +33,49 @@ export default function Component(props: Props) {
         props.handleSubmit({ type: TYPES.Calculate, data: newValues });
       }
     },
-    validate: () => {
-      // can parse formula
-      getVariables(formik.values.formula);
+    validate: (values) => {
+      const errors: FormikErrors<Calculate> = {};
+      try {
+        // can parse formula
+        getVariables(formik.values.formula);
+
+        // Validate formula
+        const result = evaluate(
+          values.formula,
+          values.samples,
+          values.defaults,
+        );
+
+        if (Number.isNaN(Number(result))) {
+          errors.formula = "Enter a formula which outputs a number";
+        }
+      } catch (error: any) {
+        errors.formula = error.message;
+      }
+      return errors;
     },
+
+    validateOnChange: false,
   });
 
+  const sampleResult = React.useMemo(() => {
+    try {
+      const result = evaluate(
+        formik.values.formula,
+        formik.values.samples,
+        formik.values.defaults,
+      );
+      // Type guard as mathjs evaluates `m` to a "Unit" object for "meter"
+      if (!Number.isNaN(Number(result))) {
```

Comment:
Is `!Number.isNaN(Number(result))` actually the same as original type guard check `typeof result === "number"` ? What's motivation for rewriting this?

Reply:
From what I could work out, the result would return a `string` if you were using a data field populated by a `NumberInput` so `result === "number"` would return `false`, although the calculation would work fine. This wasn't a problem when validating on change in the way it was, because it didn't stop you adding the calculate component, but it was a problem when I used Formik to validate on submit, which blocked it.